go-controller/pkg/metrics/ovn_db.go

```diff
@@ -261,29 +266,44 @@ func ovnDBSizeMetricsUpdater(direction, database string) {
 	metricDBSize.WithLabelValues(database).Set(float64(fileInfo.Size()))
 }
 
-func ovnE2eTimeStampUpdater(direction, database string) {
-	var stdout, stderr string
+func ovnE2eTimeStampUpdater(direction, database string, nbClient libovsdbclient.Client,
+	sbClient libovsdbclient.Client) {
+	var nbGlobal *nbdb.NBGlobal
+	var sbGlobal *sbdb.SBGlobal
 	var err error
 
-	if direction == "sb" {
-		stdout, stderr, err = util.RunOVNSbctlUnix("--if-exists", "--no-leader-only",
-			"get", "SB_Global", ".", "options:e2e_timestamp")
-	} else {
-		stdout, stderr, err = util.RunOVNNbctlUnix("--if-exists", "--no-leader-only",
-			"get", "NB_Global", ".", "options:e2e_timestamp")
-	}
-	if err != nil {
-		klog.Errorf("Failed to scrape timestamp for database %s: "+
-			"stderr (%s) (%v)", database, stderr, err)
-		return
-	}
-	if stdout != "" {
-		if value, err := strconv.ParseFloat(stdout, 64); err == nil {
+	scrapeAndSetE2ETimestampMetrics := func(options map[string]string) {
+		var v string
+		var ok bool
+
+		if v, ok = options["e2e_timestamp"]; !ok {
+			klog.Errorf("Failed to scrape timestamp from global config options")
+			return
+		}
+
+		if value, err := strconv.ParseFloat(v, 64); err == nil {
 			metricDBE2eTimestamp.WithLabelValues(database).Set(value)
 		} else {
-			klog.Errorf("Failed to parse %s e2e-timestamp value to float64 :(%v)",
-				database, err)
+			klog.Errorf("Failed to parse %s e2e-timestamp value to float64 err: %v", err)
+		}
+	}
+
+	if direction == "sb" && sbGlobal != nil {
+		if sbGlobal, err = libovsdbops.FindSBGlobal(sbClient); err != nil {
+			klog.Errorf("Failed to scrape timestamp from sb_Global table "+
+				"err: %v", database, err)
+			return
+		}
+
+		scrapeAndSetE2ETimestampMetrics(sbGlobal.Options)
+	} else if nbGlobal != nil {
+		if nbGlobal, err = libovsdbops.FindNBGlobal(nbClient); err != nil {
+			klog.Errorf("Failed to scrape timestamp from nb_Global table "+
+				"err: %v", database, err)
+			return
 		}
+
+		scrapeAndSetE2ETimestampMetrics(nbGlobal.Options)
 	} else {
```

Comment:
This is wrong, should just be else, and then internally if the nb/sbglobal lookups fail then we do `metricDBE2eTimestamp.WithLabelValues(database).Set(0)`